src/core/address.cpp

```diff
@@ -40,20 +40,22 @@ Address::Address( const std::string& street
     // Capitalize state
     for(char& c : _state) { c = std::toupper(c); }
 
+    //Grants access to class specific errors messages
+    invalid_address error_msg("",{});
```

Comment:
i did this for testing purposes so i could ensure the correct exception message is thrown during the test case for bad user inputs. I did not consider the cost too expensive since its just a few bytes and wont be done often often. i see your point about the excessive copying though, especially if loading and constructing a grip of accounts during start up. With the upcoming refactor ill put it back the way it was.

Reply:
Yeah, the cost is probably negligible but it's also a good idea to not test the content of the message but instead just test for the presence of a message. The reason being that message contents are prone to change often, and if there are tests tied to the content we will cause needless regressions.

Reply:
im pretty sure we have squashed this issues already so i wont discuss the refactor too much. suffice to say the current state of exception handling is well known by both of us and is no longer well represented by this pull request or comments within this section. also good points about the reason for why it needed to be refactored.